plugins/generator-1.19.2/forge-1.19.2/templates/livingentity/livingentity.java.ftl

```diff
@@ -272,15 +272,25 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
 		|| data.immuneToDragonBreath || data.immuneToWither>
 	@Override public boolean hurt(DamageSource source, float amount) {
 		<#if hasProcedure(data.whenMobIsHurt)>
-			<@procedureCode data.whenMobIsHurt, {
+			<#assign whenEntityHurtProcedure><#compress>
+			${data.whenMobIsHurt.getName()}Procedure.execute(<@procedureDependenciesCode data.whenMobIsHurt.getDependencies(generator.getWorkspace()), {

```

Comment:
You are hardcoding reference to execute and procedure here which should be done by utils in procedure editor. Check eg. how actionresulttype procedures handle this with call to either procedureOBJToCode or procedureOBJToInteractionResultCode

Reply:
Ok yes those are not optimized, I thought this approach also optimizes. It is good to not use optimized code in this case and go the way eg. portaltrigger.java.ftl does it